### DISCUSSION TITLE
Win10 use ImagePredictor error!

### DISCUSSION THREAD

**Discussion by 155cannon** [autogluon/autogluon#1726] [Q&A] [answered]
C:\Users\Via20220113\AppData\Local\Programs\Python\Python39\python.exe D:/Learning/AutoGluon/image_prediction/image_pred.py
`time_limit=auto` set to `time_limit=7200`.
Reset labels to [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
Randomly split train_data into train[45000]/validation[5000] splits.
The number of requested GPUs is greater than the number of available GPUs.Reduce the number to 1
Starting fit without HPO
modified configs(<old> != <new>): {
root.img_cls.model   resnet101 != resnet50
root.train.epochs    200 != 2
root.train.early_stop_baseline 0.0 != -inf
root.train.early_stop_max_value 1.0 != inf
root.train.batch_size 32 != 16
root.train.early_stop_patience -1 != 10
root.misc.num_workers 4 != 8
root.misc.seed       42 != 750
}
Saved config to D:\Learning\AutoGluon\image_prediction\594dfaaa\.trial_0\config.yaml
Downloading: "https://github.com/rwightman/pytorch-image-models/releases/download/v0.1-rsb-weights/resnet50_a1_0-14fe96d1.pth" to C:\Users\Via20220113/.cache\torch\hub\checkpoints\resnet50_a1_0-14fe96d1.pth
Finished, total runtime is 773.64 s
{ 'best_config': { 'batch_size': 16,
                   'dist_ip_addrs': None,
                   'early_stop_baseline': -inf,
                   'early_stop_max_value': inf,
                   'early_stop_patience': 10,
                   'epochs': 2,
                   'final_fit': False,
                   'gpus': [0],
                   'lr': 0.01,
                   'model': 'resnet50',
                   'ngpus_per_trial': 8,
                   'nthreads_per_trial': 128,
                   'num_workers': 8,
                   'searcher': 'random',
                   'seed': 750,
                   'time_limits': 7200},
  'total_time': 773.6272704601288,
  'train_acc': -1,
  'valid_acc': -1}
Traceback (most recent call last):
  File "D:\Learning\AutoGluon\image_prediction\image_pred.py", line 32, in <module>
    predictor.fit(train_data, hyperparameters={'epochs': 2})
  File "C:\Users\Via20220113\AppData\Local\Programs\Python\Python39\lib\site-packages\autogluon\vision\configs\presets_configs.py", line 18, in _call
    return f(*gargs, **gkwargs)
  File "C:\Users\Via20220113\AppData\Local\Programs\Python\Python39\lib\site-packages\autogluon\vision\predictor\predictor.py", line 419, in fit
    self._classifier = task.fit(train_data, tuning_data, 1 - holdout_frac, random_state)
  File "C:\Users\Via20220113\AppData\Local\Programs\Python\Python39\lib\site-packages\autogluon\vision\_gluoncv\image_classification.py", line 421, in fit
    raise RuntimeError(f'Unexpected error happened during fit: {pprint.pformat(results, indent=2)}')
RuntimeError: Unexpected error happened during fit: { 'args': "{'img_cls': {'model': 'resnet50', 'pretrained': True, "
          "'global_pool_type': None}, 'data': {'img_size': None, 'input_size': "
          "None, 'crop_pct': 0.99, 'mean': None, 'std': None, 'interpolation': "
          "'', 'validation_batch_size_multiplier': 1}, 'optimizer': {'opt': "
          "'sgd', 'opt_eps': None, 'opt_betas': None, 'momentum': 0.9, "
          "'weight_decay': 0.0001, 'clip_grad': None, 'clip_mode': 'norm'}, "
          "'train': {'batch_size': 16, 'sched': 'step', 'lr': 0.01, "
          "'lr_noise': None, 'lr_noise_pct': 0.67, 'lr_noise_std': 1.0, "
          "'lr_cycle_mul': 1.0, 'lr_cycle_limit': 1, 'transfer_lr_mult': 0.01, "
          "'output_lr_mult': 0.1, 'warmup_lr': 0.0001, 'min_lr': 1e-05, "
          "'epochs': 2, 'start_epoch': 0, 'decay_epochs': 30, 'warmup_epochs': "
          "3, 'cooldown_epochs': 10, 'patience_epochs': 10, 'decay_rate': 0.1, "
          "'bn_momentum': None, 'bn_eps': None, 'sync_bn': False, "
          "'early_stop_patience': 10, 'early_stop_min_delta': 0.001, "
          "'early_stop_baseline': -inf, 'early_stop_max_value': inf}, "
          "'augmentation': {'no_aug': False, 'scale': (0.08, 1.0), 'ratio': "
          "(0.75, 1.3333333333333333), 'hflip': 0.5, 'vflip': 0.0, "
          "'color_jitter': 0.4, 'auto_augment': None, 'mixup': 0.0, 'cutmix': "
          "0.0, 'cutmix_minmax': None, 'mixup_prob': 1.0, 'mixup_switch_prob': "
          "0.5, 'mixup_mode': 'batch', 'mixup_off_epoch': 0, 'smoothing': 0.1, "
          "'train_interpolation': 'random', 'drop': 0.0, 'drop_path': None, "
          "'drop_block': None}, 'model_ema': {'model_ema': True, "
          "'model_ema_force_cpu': False, 'model_ema_decay': 0.9998}, 'misc': "
          "{'seed': 750, 'log_interval': 50, 'num_workers': 8, 'save_images': "
          "False, 'amp': False, 'apex_amp': False, 'native_amp': False, "
          "'pin_mem': False, 'prefetcher': False, 'eval_metric': 'top1', "
          "'tta': 0, 'use_multi_epochs_loader': False, 'torchscript': False}, "
          "'gpus': [0]}",
  'time': 773.6272704601288,
  'traceback': 'Traceback (most recent call last):\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\autogluon\\vision\\_gluoncv\\image_classification.py", '
               'line 191, in _train_image_classification\n'
               '    result = estimator.fit(train_data=train_data, '
               'val_data=val_data, time_limit=wall_clock_tick-tic)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\gluoncv\\auto\\estimators\\base_estimator.py", '
               'line 152, in fit\n'
               '    self._init_network()\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\gluoncv\\auto\\estimators\\torch_image_classification\\torch_image_classification.py", '
               'line 512, in _init_network\n'
               '    self.net = create_model(\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\timm\\models\\factory.py", '
               'line 74, in create_model\n'
               '    model = create_fn(pretrained=pretrained, **kwargs)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\timm\\models\\resnet.py", '
               'line 764, in resnet50\n'
               "    return _create_resnet('resnet50', pretrained, "
               '**model_args)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\timm\\models\\resnet.py", '
               'line 694, in _create_resnet\n'
               '    return build_model_with_cfg(\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\timm\\models\\helpers.py", '
               'line 465, in build_model_with_cfg\n'
               '    load_pretrained(\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\timm\\models\\helpers.py", '
               'line 189, in load_pretrained\n'
               '    state_dict = load_state_dict_from_url(pretrained_url, '
               "progress=progress, map_location='cpu')\n"
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\torch\\hub.py", '
               'line 586, in load_state_dict_from_url\n'
               '    download_url_to_file(url, cached_file, hash_prefix, '
               'progress=progress)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\site-packages\\torch\\hub.py", '
               'line 474, in download_url_to_file\n'
               '    buffer = u.read(8192)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\http\\client.py", '
               'line 463, in read\n'
               '    n = self.readinto(b)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\http\\client.py", '
               'line 507, in readinto\n'
               '    n = self.fp.readinto(b)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\socket.py", '
               'line 704, in readinto\n'
               '    return self._sock.recv_into(b)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\ssl.py", '
               'line 1241, in recv_into\n'
               '    return self.read(nbytes, buffer)\n'
               '  File '
               '"C:\\Users\\Via20220113\\AppData\\Local\\Programs\\Python\\Python39\\lib\\ssl.py", '
               'line 1099, in read\n'
               '    return self._sslobj.read(len, buffer)\n'
               'ConnectionResetError: [WinError 10054] 远程主机强迫关闭了一个现有的连接。\n',
  'train_acc': -1,
  'valid_acc': -1}

进程已结束,退出代码1
My PC environment:

autogluon               0.4.0
autogluon.common        0.4.0
autogluon-contrib-nlp   0.0.1b20220208
autogluon.core          0.4.0
autogluon.features      0.4.0
autogluon.tabular       0.4.0
autogluon.text          0.4.0
autogluon.vision        0.4.0

TabularPredictor is OK in my PC. ImagePredictor is running with 'ConnectionResetError: [WinError 10054]'. Is this just network error?

**Comment by Innixma** [comment, marked as answer]
Closing as duplicate of #1727
